zebra/zebra_vxlan.c

```diff
@@ -4212,6 +4203,10 @@ static void process_remote_macip_add(vni_t vni,
 			}
 		}
 
+		/* Remove local MAC from BGP. */
+		if (CHECK_FLAG(mac->flags, ZEBRA_MAC_LOCAL))
```

Comment:
from the explanation at #5, shouldn't bgp ignore the local mac update as it has sequence as 1 against the remote mac update sequence number which is 2 ? So why do we need this explicit delete ?

Reply:
BGP may not pick the path as best but it will leave it the VNI and global routing tables. As zebra is the owner of local routes it must explicitly delete all entries that it added including the ones it decided to modify from local=>remote otherwise bgpd could end up falling out of sync with zebra (which is actually the primary problem that was seen in a POD).